src/internal_events/kafka.rs

```diff
@@ -57,3 +57,887 @@ impl InternalEvent for KafkaKeyExtractionFailed<'_> {
+                app_offset: partition.app_offset,
+                stored_offset: partition.stored_offset,
+                committed_offset: partition.committed_offset,
+                eof_offset: partition.eof_offset,
+                lo_offset: partition.lo_offset,
+                hi_offset: partition.hi_offset,
+                ls_offset: partition.ls_offset,
+                consumer_lag: partition.consumer_lag,
+                txmsgs: partition.txmsgs,
+                txbytes: partition.txbytes,
+                rxmsgs: partition.rxmsgs,
+                rxbytes: partition.rxbytes,
+                msgs: partition.msgs,
+                rx_ver_drops: partition.rx_ver_drops,
+                msgs_inflight: partition.msgs_inflight,
+                next_ack_seq: partition.next_ack_seq,
+                next_err_seq: partition.next_err_seq,
+                acked_msgid: partition.acked_msgid,
+            }
+        }
+    }
+
+    impl From<rdkafka::statistics::ConsumerGroup> for ConsumerGroup {
+        fn from(cgrp: rdkafka::statistics::ConsumerGroup) -> Self {
+            Self {
+                state: cgrp.state,
+                stateage: cgrp.stateage,
+                join_state: cgrp.join_state,
+                rebalance_age: cgrp.rebalance_age,
+                rebalance_cnt: cgrp.rebalance_cnt,
+                rebalance_reason: cgrp.rebalance_reason,
+                assignment_size: cgrp.assignment_size,
+            }
+        }
+    }
+
+    impl From<rdkafka::statistics::ExactlyOnceSemantics> for ExactlyOnceSemantics {
+        fn from(eos: rdkafka::statistics::ExactlyOnceSemantics) -> Self {
+            Self {
+                idemp_state: eos.idemp_state,
+                idemp_stateage: eos.idemp_stateage,
+                txn_state: eos.txn_state,
+                txn_stateage: eos.txn_stateage,
+                txn_may_enq: eos.txn_may_enq,
+                producer_id: eos.producer_id,
+                producer_epoch: eos.producer_epoch,
+                epoch_cnt: eos.epoch_cnt,
+            }
+        }
+    }
+
+    #[cfg(test)]
+    mod tests {
+        use super::*;
+        use serde_json;
+
+        #[test]
+        fn test_statistics_json_roundtrip() {
+            let rdkafka_statistics: rdkafka::Statistics = serde_json::from_str(EXAMPLE).unwrap();
+            let statistics: Statistics = rdkafka_statistics.into();
+            let json = serde_json::to_string(&statistics).unwrap();
+
+            let original_json: serde_json::Value = serde_json::from_str(EXAMPLE).unwrap();
+            let roundtrip_json: serde_json::Value = serde_json::from_str(&json).unwrap();
+
+            assert_eq!(original_json, roundtrip_json);
+        }
+
+        // Example from https://github.com/edenhill/librdkafka/wiki/Statistics with small modifications:
+        // - Removed "commited_offset" fields (redundant/misspelled).
+        // - Removed statistics.broker[$broker].req fields (not provided by `rdkafka`).
```

Comment:
Opened an upstream issue for that: https://github.com/fede1024/rust-rdkafka/issues/334 and might submit a PR there.

Reply:
https://github.com/fede1024/rust-rdkafka/pull/335 has been merged, but not released yet. If there's one out in time, I can bump the dependency and add the field to the test again.

Reply:
No longer relevant, since we don't need to serialize a statistics JSON anymore.